service-base/src/test/java/org/eclipse/hono/service/AbstractProtocolAdapterBaseTest.java

```diff
@@ -589,30 +591,33 @@ public void testCheckConnectionLimitFailsIfMessageLimitIsReached(final VertxTest
         }));
     }
 
-    private AbstractProtocolAdapterBase<ProtocolAdapterProperties> newProtocolAdapter(final ProtocolAdapterProperties props) {
+    private AbstractProtocolAdapterBase<ProtocolAdapterProperties, Metrics> newProtocolAdapter(
+            final ProtocolAdapterProperties props) {
 
         return newProtocolAdapter(props, ADAPTER_NAME);
     }
 
-    private AbstractProtocolAdapterBase<ProtocolAdapterProperties> newProtocolAdapter(final ProtocolAdapterProperties props, final String typeName) {
+    private AbstractProtocolAdapterBase<ProtocolAdapterProperties, Metrics> newProtocolAdapter(
+            final ProtocolAdapterProperties props, final String typeName) {
         return newProtocolAdapter(props, typeName, start -> {});
     }
 
-    private AbstractProtocolAdapterBase<ProtocolAdapterProperties> newProtocolAdapter(
+    private AbstractProtocolAdapterBase<ProtocolAdapterProperties, Metrics> newProtocolAdapter(
             final ProtocolAdapterProperties props,
             final String typeName,
             final Handler<Void> startupHandler) {
         return newProtocolAdapter(props, typeName, startupHandler, null, null);
     }
 
-    private AbstractProtocolAdapterBase<ProtocolAdapterProperties> newProtocolAdapter(
+    private AbstractProtocolAdapterBase<ProtocolAdapterProperties, Metrics> newProtocolAdapter(
             final ProtocolAdapterProperties props,
             final String typeName,
             final Handler<Void> startupHandler,
             final Handler<Void> commandConnectionEstablishedHandler,
             final Handler<Void> commandConnectionLostHandler) {
 
-        final AbstractProtocolAdapterBase<ProtocolAdapterProperties> result = new AbstractProtocolAdapterBase<>() {
+        final AbstractProtocolAdapterBase<ProtocolAdapterProperties, Metrics> result = new AbstractProtocolAdapterBase<>(
+                mock(NoopBasedMetrics.class)) {
```

Comment:
we usually do not mock classes but interfaces only. Is there a particular reason why you do not want to mock `Metrics` instead?

Reply:
The reason is that I don't need to mock the event bus for the NoopBasedMetrics, since it is not used in that case. But I thinking again, this reason is not good enough, so I'll change that.